lib/elixir/lib/access.ex

```diff
@@ -669,7 +670,17 @@ defmodule Access do
     end
   end
 
-  defp get_and_update_at([head | rest], index, next, updates) do
+  defp get_and_update_at(list, index, next, updates)
+       when index < 0 and length(list) + index >= 0 do
```

Comment:
This is computing the length twice (here and below). I would recommend to move this to inside the function body and  merge this clause with the one below so we have to do it only once.

Reply:
hi, thanks for the quick feedback. Update the PR  merging the the functions and removing the length calculation from the guard